### ISSUE TITLE
[MPIR-142] Dependency report fails to generate if dependencies contain com.ibm.icu:icu4j:2.6.1

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-project-info-reports-plugin#439]
**[Marat Radchenko](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slonopotamus)** opened **[MPIR-142](https://issues.apache.org/jira/browse/MPIR-142?redirect=false)** and commented

Possibly caused by https://issues.apache.org/bugzilla/show_bug.cgi?id=28069

```
[WARNING] Unable to process class com/ibm/icu/impl/data/LocaleElements_zh__PINYIN.class in JarAnalyzer File /home/marat/.m2/repository/com/ibm/icu/icu4j/2.6.1/icu4j-2.6.1.jar
org.apache.bcel.classfile.ClassFormatException: Invalid byte tag in constant pool: 60
	at org.apache.bcel.classfile.Constant.readConstant(Constant.java:146)
	at org.apache.bcel.classfile.ConstantPool.<init>(ConstantPool.java:67)
	at org.apache.bcel.classfile.ClassParser.readConstantPool(ClassParser.java:222)
	at org.apache.bcel.classfile.ClassParser.parse(ClassParser.java:136)
	at org.apache.maven.shared.jar.classes.JarClassesAnalysis.analyze(JarClassesAnalysis.java:92)
	at org.apache.maven.report.projectinfo.dependencies.Dependencies.getJarDependencyDetails(Dependencies.java:285)
	at org.apache.maven.report.projectinfo.dependencies.renderer.DependenciesRenderer.hasSealed(DependenciesRenderer.java:1278)
	at org.apache.maven.report.projectinfo.dependencies.renderer.DependenciesRenderer.renderSectionDependencyFileDetails(DependenciesRenderer.java:423)
	at org.apache.maven.report.projectinfo.dependencies.renderer.DependenciesRenderer.renderBody(DependenciesRenderer.java:268)
	at org.apache.maven.reporting.AbstractMavenReportRenderer.render(AbstractMavenReportRenderer.java:65)
	at org.apache.maven.report.projectinfo.DependenciesReport.executeReport(DependenciesReport.java:239)
	at org.apache.maven.reporting.AbstractMavenReport.generate(AbstractMavenReport.java:90)
	at org.apache.maven.plugins.site.ReportDocumentRenderer.renderDocument(ReportDocumentRenderer.java:139)
	at org.apache.maven.doxia.siterenderer.DefaultSiteRenderer.renderModule(DefaultSiteRenderer.java:269)
	at org.apache.maven.doxia.siterenderer.DefaultSiteRenderer.render(DefaultSiteRenderer.java:101)
	at org.apache.maven.plugins.site.SiteMojo.renderLocale(SiteMojo.java:133)
	at org.apache.maven.plugins.site.SiteMojo.execute(SiteMojo.java:100)
	at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:451)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:558)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalWithLifecycle(DefaultLifecycleExecutor.java:499)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:478)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:330)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:291)
	at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:142)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:287)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
	at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
	at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:375)

```

---

**Affects:** 2.1

**Issue Links:**
- [MPLUGIN-221](https://issues.apache.org/jira/browse/MPLUGIN-221) DefaultMojoAnnotationsScanner fails to scan some dependencies

9 votes, 7 watchers

**Comment by jira-importer** [comment]
**[Marat Radchenko](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slonopotamus)** commented

Woops, icu4j version is 2.6.1, not 3.6.1

**Comment by jira-importer** [comment]
**[Stefan Seidel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=seidler)** commented

Workaround for the time being is to disable the dependencies report:

```
<reporting>
  <plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-project-info-reports-plugin</artifactId>
      <reportSets>
        <reportSet>
          <reports>
            <report>index</report>
            <report>cim</report>
            ...
          </reports>
        </reportSet>
      </reportSets>
    </plugin>
  </plugins>
</reporting>
```

(just make sure you do not include "dependencies" as a report.)

**Comment by jira-importer** [comment]
**[phil](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ph.z)** commented

I know it's sort of a brutal method but it worked for me: simply delete the LocaleElements_zh__PINYIN.class file from the icu4j-2.6.1.jar archive. Unless you are a "Chinese PINYIN speaker" it shouldn't matter. Without this file the dependency report generated fine and the project still got built without errors. Just put that "patched" icu4j-2.6.1.jar file in your local mvn repo and you're done.

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

It is a warning message and the report is generated as well with icu4j

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

FYI, it's a known bug in icu4j 2.6: see [ticket #6505](http://icu-project.org/trac/ticket/6505), which shows the problem wihtout Maven or bcel but directly a JVM
marked as fixed in icu4j 2.8: see [ticket #3209](http://icu-project.org/trac/ticket/3209)
